src/algorithms/calorimetry/ImagingTopoCluster.h

```diff
@@ -12,6 +12,7 @@
 #pragma once
```

Comment:
Regarding the attribution comment at the start of the file: I think if we have top-level comments with attribution like this we should keep them up-to-date. In this case you are modifying the algorithm so that should be mentioned and attributed. Or we should at least mention that the "first version" was based on the paper IMO.

Reply:
The algorithm isn't actually getting modified. The only thing that's modified is the choice of containers (vector -> set), the way of encoding that hits have been picked already (set membership), and the way groups are grown. I don't see this as a different algorithm. This should return the 100% identical proto-cluster hits (possibly in a different order).

Even sorting hits at first, and then modifying the selection loop by not considering hits that will clearly not match the requirements is an optimization, not a change to the algorithm. Maybe that's just semantics.

Reply:
26790cf0ccac6b0941ff80580daccdd32c09df14